### ISSUE TITLE
Class name for an actor metric occurs twice in the metric name

### ISSUE THREAD

**Issue by provegard** [erikvanoosten/metrics-scala#75]
I see something like:

```
name=com.foo.bar.MyActor.com.foo.bar.MyActor.receiveCounter
```

I suppose this is because the actor traits use the class name in the metric name, which then is appended onto the base name, which also is the class name:

```
trait ReceiveCounterActor extends Actor { self: InstrumentedBuilder =>

  def receiveCounterName: String = MetricName(getClass).append("receiveCounter").name
  lazy val counter: Counter = metrics.counter(receiveCounterName)
```

`counter` calls `metricNameFor` which is implemented as: `baseName.append(name, scope).name`, and the base name comes from:

```
trait BaseBuilder {
  /** The base name for all metrics created from this builder. */
  lazy val metricBaseName = MetricName(getClass)
}
```

**Comment by erikvanoosten** [comment]
Hi Per. Thanks for reporting! We can fix this, though it will not be backward compatible for existing users.

I think I'll grab the opportunity to make another change: let the method accept a `Timer` instead of a `String`.

**Comment by erikvanoosten** [comment]
Wait, never mind that last sentence.

**Comment by erikvanoosten** [comment]
@scullxbones would you like to take a look at my changes above?

**Comment by scullxbones** [comment]
Yep looks good to me - might want to update your local git config, github is having a hard time recognizing the commit as coming from you.

**Comment by erikvanoosten** [comment]
I see. Thanks for the warning.

**Comment by erikvanoosten** [comment]
Fixed in 3.5.4.

**Comment by provegard** [comment]
Excellent, thanks!
